dist/docker/entrypoint.sh

```diff
@@ -1,21 +1,27 @@
 #!/bin/sh
 
-if [ ! -f /config/qBittorrent/qBittorrent.conf ]; then
-    mkdir -p /config/qBittorrent/
-    cat << EOF > /config/qBittorrent/qBittorrent.conf
+profilePath="/config"
+qbtConfigFile="$profilePath/qBittorrent/config/qBittorrent.conf"
+
+if [ ! -f "$qbtConfigFile" ]; then
+    mkdir -p "$(dirname $qbtConfigFile)"
+    cat << EOF > "$qbtConfigFile"
 [BitTorrent]
 Session\DefaultSavePath=/downloads
+Session\Port=6881
 Session\TempPath=/downloads/temp
 
 [LegalNotice]
 Accepted=false
 EOF
-fi
 
-if [ "$LEGAL" = "accept" ] ; then
-    sed -i '/^\[LegalNotice\]$/{$!{N;s|\(\[LegalNotice\]\nAccepted=\).*|\1true|}}' /config/qBittorrent/qBittorrent.conf
-else
-    sed -i '/^\[LegalNotice\]$/{$!{N;s|\(\[LegalNotice\]\nAccepted=\).*|\1false|}}' /config/qBittorrent/qBittorrent.conf
+    if [ "$QBT_EULA" = "accept" ]; then
+        sed -i '/^\[LegalNotice\]$/{$!{N;s|\(\[LegalNotice\]\nAccepted=\).*|\1true|}}' "$qbtConfigFile"
+    else
+        sed -i '/^\[LegalNotice\]$/{$!{N;s|\(\[LegalNotice\]\nAccepted=\).*|\1false|}}' "$qbtConfigFile"
+    fi
 fi
 
-HOME="/config" XDG_CONFIG_HOME="/config" XDG_DATA_HOME="/data" qbittorrent-nox --webui-port=$WEBUI_PORT
+exec qbittorrent-nox \
```

Comment:
Will `exec` properly handle signal forwarding and cleanup of zombie processes? If not, we can use [tini](https://github.com/krallin/tini).

Reply:
`exec` is not a program itself, it replaces the shell script process with `qbittorrent-nox`, so the latter will need to handle those tasks you mentioned.
Since I don't think qbt can handle zombie processes properly, it should be a good idea to use `tini`.